includes/class-wc-regenerate-images-request.php

```diff
@@ -61,17 +59,28 @@ protected function task( $item ) {
 			return false;
 		}
 
+		$attachment_id      = absint( $item['attachment_id'] );
+		$last_attachment_id = absint( get_option( 'woocommerce_last_image_regen_id', 0 ) );
+
+		// If the current task matches the last task there may have been an error which prevented the task from completing; skip.
+		if ( $last_attachment_id === $attachment_id ) {
+			return false;
+		}
```

Comment:
I think this check should solve the issues, but may theoretically end up with only regenerating most of the images. 

Looking through how the background process class works, it checks after each task (one image regeneration) to see if the time limit (20s) has been exceeded, so if we're getting server timeouts that means it is timing out **during** image processing.

This can happen like so: If it takes 16s to generate all thumbnails for one image and there is a 30s PHP process limit: 
- The first image regeneration will work properly (6s left of the 20s background processing limit)
- The second image regeneration will cause a server timeout (32s used of 30s PHP process limit)
- The process starts back up and the second image gets skipped. This pattern repeats.

One other potential solution would be to break out each image resize into a separate task (e.g. generate thumbnail size, then background processor can check if time limit exceeded, then generate thumbnail 2x size, then background processor can check if time limit exceeded, etc.). Until proven otherwise, though, the solution in this PR is good and that change would be in addition to what you have here.

Reply:
I like your suggestion on decreasing the default limit, and I am pretty sure this is exactly what is happening here. First one or two resizes complete then another fires off and times out due to reaching the overall time limit.

That said I do not think it is a bug, but a feature of the background processing library as the failed task will re-queue and get picked up in the next batch run, it just obviously ads something to your logs to indicate that the script timed out.